### PR TITLE
Install and configuration of paper_trail for auditing of Hold records

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds the paper_trail gem to the application, enabling it on the Holds model to enable an audit trail of all holds placed and removed from a Thesis. No other model is affected by this change.

In adding this gem, we are using the `--with-changes` flag in order to track the actual changes made. Additionally, the hook for enabling recording the user ID who makes a change has been added at the app controller level, in order to not lose this information stream should changes end up happening elsewhere. This enables the `whodunnit` method wherever we need it.

The a11y tests have not been performed on these changes, as there are no UI impacts on any of these code changes.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

YES

#### Includes new or updated dependencies?

YES

### CODE REVIEW

**Review by JPrevost — COMMENTED** [review]
I'd like us to chat in #engx on slack about the question I put in about where to put the `set_paper_trail_whodunnit` and a minor test name change would be nice but technically isn't required. Looks good though!

File: app/controllers/admin/holds_controller.rb [hunk 1]
```diff
@@ -42,5 +42,6 @@ class HoldsController < Admin::ApplicationController
 
     # See https://administrate-prototype.herokuapp.com/customizing_controller_actions
     # for more information
+    before_action :set_paper_trail_whodunnit
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/controllers/admin/holds_controller.rb]: This makes sense - I've moved the flag up to the app conotroller.

File: test/integration/admin_dashboard_test.rb [hunk 2]
```diff
@@ -475,6 +475,16 @@ def teardown
     assert_select "select#hold_status", text: "active\nexpired\nreleased"
   end
 
+  test 'editing a hold while authenticated stores the editor identity' do
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply test/integration/admin_dashboard_test.rb]: Yeah, that's fair - I've renamed the test.

**Review by JPrevost — APPROVED** [review]
My outstanding question from slack about whether we should store the user_id or the user name is non blocking. We can think about that and adjust. This work is :shipit:

**Comment by matt-bernhardt** [comment]
Merging this now, we will get stakeholder approval once these changes are manifest in a UI somewhere.

**Pull request merged**